python/pyarrow/tests/test_array.py

```diff
@@ -2666,6 +2666,31 @@ def test_array_masked():
     assert arr.type == pa.int64()
 
 
+def test_binary_array_masked():
+    # ARROW-12431
+    masked_basic = pa.array([b'\x05'], type=pa.binary(1),
+                            mask=np.array([False]))
+    assert pa.array([b'\x05']).to_pylist() == masked_basic.to_pylist()
```

Comment:
```suggestion
    assert [b'\x05'] == masked_basic.to_pylist()
```

I would compare it here to the known result, since it's a simple result (and to not rely on another conversion path)

Reply:
👍